### DISCUSSION TITLE
First item of custom menu selected after closing rofi menu w/out input

### DISCUSSION THREAD

**Discussion by arilebedey** [davatorium/rofi#1563] [General]
I know this issue probably has a very easy solution but even after searching, I could not find it..

Here is my bare-bones script: 
```
#!/bin/sh

link="Useful menu" 
ch="Chrome"

sel=$(echo "${ch}" | rofi -dmenu -p "${link}")

if [[ "${sel}" == "$ch" ]]; then
        chromium
fi
```
I am trying to find a way to cancel the menu without rofi issuing the `chromium` command.. 

I have tried passing the variable "MENU_CUSTOM_INPUT" mentioned in a previous commit (cited below) that closed a similar issue, but it did not work..

Also : I have added "kb-cancel: "Escape";" to the configuration portion of `config.rasi`..

Above mentioned commit:    
https://github.com/davatorium/rofi/issues/322#issuecomment-174425417_

**Comment by DaveDavenport** [comment]
This is very confusing.

MENU_CUSTOM_INPUT is internal to rofi, not a variable to be set? 

Further more rofi does launch chromium, you do in your script.

if you 'cancel' the menu (default via escape) it returns an empty string, then in the above script it should not launch chromium. (and does not if I test it here after replacing sh -> bash).

your shebang seems to be wrong, this needs to be bash?

**Reply by arilebedey** [reply]
Ok.. I have changed the shebang to `#!/bin/bash` and the error still persists..

**Reply by DaveDavenport** [reply]
If you press escape, rofi returns an empty string (check by printing sel) and your function should not match.
The script works here.

But hard to debug without more information, but this seems more a scripting issue.
What rofi version do you run?

**Reply by arilebedey** [reply]
It was a scripting error in that I had unused if statements such as:
```
if [[ "${sel}" == "$ym" ]]; then
        chromium "https://music.youtube.com"
fi
```
..in this script..

and trying to use in-existing variables in another script with the same issue..
...👍

Thank you for the assist.. 😃
